From eaadfe73764de81f73d703da5c91bf4a9a423875 Mon Sep 17 00:00:00 2001
From: mssx86 <mss@waifu.club>
Date: Sun, 23 Feb 2020 11:57:53 +0300
Subject: [PATCH] upd: grayscale colors.

---
 xmand/not_double_r_col.c | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/xmand/not_double_r_col.c b/xmand/not_double_r_col.c
index 5d07fee..30bf335 100644
--- a/xmand/not_double_r_col.c
+++ b/xmand/not_double_r_col.c
@@ -25,36 +25,36 @@ unsigned long ndr ( uint8_t height )
     unsigned long cpixel = 0;
 
     if ( height < 16 ) {
-        cpixel = linear_inter( height, (uint32_t)0x8a410a,
-                                       (uint32_t)0xe62063,
+        cpixel = linear_inter( height, (uint32_t)0x101010,
+                                       (uint32_t)0x252525,
                                        (uint8_t)0, (uint8_t)15);
     } else if ( ( height > 15 ) && ( height < 32 ) ) {
-        cpixel = linear_inter( height, (uint32_t)0xe62063,
-                                       (uint32_t)0x3af8ff,
+        cpixel = linear_inter( height, (uint32_t)0x252525,
+                                       (uint32_t)0x464646,
                                        (uint8_t)16, (uint8_t)31);
     } else if ( ( height > 31 ) && ( height < 128 ) ) {
-        cpixel = linear_inter( height, (uint32_t)0x3af8ff,
-                                       (uint32_t)0xb7307b,
+        cpixel = linear_inter( height, (uint32_t)0x464646,
+                                       (uint32_t)0x525252,
                                        (uint8_t)32, (uint8_t)127);
     } else if ( ( height > 127 ) && ( height < 160 ) ) {
-        cpixel = linear_inter( height, (uint32_t)0xb7307b,
-                                       (uint32_t)0xecff3a,
+        cpixel = linear_inter( height, (uint32_t)0x525252,
+                                       (uint32_t)0xababab,
                                        (uint8_t)128, (uint8_t)159);
     } else if ( ( height > 159 ) && ( height < 192 ) ) {
-        cpixel = linear_inter( height, (uint32_t)0xecff3a,
-                                       (uint32_t)0x721a1a,
+        cpixel = linear_inter( height, (uint32_t)0xababab,
+                                       (uint32_t)0xffffff,
                                        (uint8_t)160, (uint8_t)191);
     } else if ( ( height > 191 ) && ( height < 224 ) ) {
-        cpixel = linear_inter( height, (uint32_t)0x721a1a,
-                                       (uint32_t)0x00ff00,
+        cpixel = linear_inter( height, (uint32_t)0xffffff,
+                                       (uint32_t)0x7c7c7c,
                                        (uint8_t)192, (uint8_t)223);
     } else if ( ( height > 223 ) && ( height < 240 ) ) {
-        cpixel = linear_inter( height, (uint32_t)0x00ff00,
-                                       (uint32_t)0xff00ff,
+        cpixel = linear_inter( height, (uint32_t)0x7c7c7c,
+                                       (uint32_t)0x999999,
                                        (uint8_t)224, (uint8_t)239);
     } else if ( height > 239 ) {
-        cpixel = linear_inter( height, (uint32_t)0xff00ff,
-                                       (uint32_t)0xffffff,
+        cpixel = linear_inter( height, (uint32_t)0x999999,
+                                       (uint32_t)0xa0a0a0,
                                        (uint8_t)240, (uint8_t)255);
     }
 
-- 
