--- filetimes_orig.c	2022-04-28 20:55:15.390779000 +0200
+++ filetimes.c	1969-12-31 23:59:59.000000000 +0100
@@ -1,4 +1,3 @@
-
 /*
  * filetimes.c report the various essential UNIX timestamps for a file
  * Copyright (C) Dennis Clarke 2019
@@ -36,6 +35,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
+#include <string.h>
 #include <unistd.h>
 #include <errno.h>
 #include <sys/types.h>
@@ -44,7 +44,7 @@
 #include <fcntl.h>
 
 #define VERBOSE 1
-int sysinfo(int verbose);
+// int sysinfo(int verbose);
 
 int main(int argc, char **argv)
 {
@@ -53,7 +53,7 @@
 
     FILE *fp;
     char *line = NULL;
-    char *c_time_string, buff[64];
+    char *c_time_string;
     size_t line_length, line_count, char_count;
     size_t fwrite_ret, linecap = 0;
     ssize_t linelen;
@@ -71,7 +71,7 @@
         return EXIT_FAILURE;
     }
 
-    setlocale ( LC_ALL, "POSIX" );
+    setlocale ( LC_ALL, "C" );
     if ( clock_gettime( CLOCK_REALTIME, &start_tv ) == -1 ) {
         perror( "clock gettime" );
         return EXIT_FAILURE;
@@ -79,7 +79,7 @@
 
     /* note that we no longer will get an ENOMEM from the
      * sysctlbyname call.  See sysinfo.c for details. */
-    sysinfo(VERBOSE);
+    // sysinfo(VERBOSE);
 
     c_time_string = ctime( &start_tv.tv_sec );
 
@@ -90,23 +90,14 @@
         fprintf (stderr,"     : three UNIX times of the file are :\n");
 
         /* access time */
-        sprintf(buff,"     : ctime(&buffer.st_atime) = %s",
-                                        ctime(&status_buffer.st_atime));
-        buff[57] = '\0';
-        fprintf (stderr,"%s\n", buff);
-
+        fprintf(stderr,"     : ctime(&buffer.st_atime) = %s",
+                               ctime(&status_buffer.st_atime));
         /* modification time */
-        sprintf(buff,"     :              .st_mtime) = %s",
-                                        ctime(&status_buffer.st_mtime));
-        buff[57] = '\0';
-        fprintf (stderr,"%s\n", buff);
-
+        fprintf(stderr,"     :              .st_mtime) = %s",
+                               ctime(&status_buffer.st_mtime));
         /* creation time */
-        sprintf(buff,"     :              .st_ctime) = %s",
-                                        ctime(&status_buffer.st_ctime));
-        buff[57] = '\0';
-        fprintf (stderr,"%s\n", buff);
-
+        fprintf(stderr,"     :              .st_ctime) = %s",
+                               ctime(&status_buffer.st_ctime));
     } else {
         /* check for a pile of things that could have gone
          * wrong */
@@ -121,7 +112,7 @@
                 fprintf ( stderr, "ERROR : EBADF\n" );
                 break;
             default :
-                fprintf ( stderr, "ERROR : something bad happened.\n" );
+                fprintf ( stderr, "ERROR : %s\n", strerror(errno) );
         }
         perror("ERROR ");
         return EXIT_FAILURE;
@@ -191,7 +182,7 @@
                      * because we did not get a valid char. */
                     line[char_count] = '\0';
                     line_count += 1;
-                    printf("%-6i    %-4i    \"%s\"\n",line_count,
+                    printf("%-6zu    %-4zu    \"%s\"\n",line_count,
                                                       char_count,
                                                       line);
 
@@ -214,7 +205,7 @@
             if ( line[char_count - 1] == '\n' ) {
                 line[char_count - 1] = '\0';
                 line_count += 1;
-                printf("%-6i    %-4i    \"%s\"\n",line_count,char_count,line);
+                printf("%-6zu    %-4zu    \"%s\"\n",line_count,char_count,line);
                 char_count = 0;
                 line[0] = '\0';
             }
@@ -229,7 +220,7 @@
         perror("FAIL ");
         return EXIT_FAILURE;
     }
-    fprintf(stderr,"\n\nINFO : ftell_pos = %-6i\n", ftell_pos);
+    fprintf(stderr,"\n\nINFO : ftell_pos = %-6li\n", ftell_pos);
 
     if ( ferror(fp) != 0 ) {
         fprintf(stderr,"\nWARN : some sort of an error occured??\n");
@@ -252,23 +243,16 @@
         fprintf (stderr,"\nINFO : three UNIX times are now :\n");
 
         /* access time */
-        sprintf(buff,"     : ctime(&buffer.st_atime) = %s",
-                                        ctime(&status_buffer.st_atime));
-        buff[57] = '\0';
-        fprintf (stderr,"%s\n", buff);
+        fprintf(stderr,"     : ctime(&buffer.st_atime) = %s",
+                               ctime(&status_buffer.st_atime));
 
         /* modification time */
-        sprintf(buff,"     :              .st_mtime) = %s",
-                                        ctime(&status_buffer.st_mtime));
-        buff[57] = '\0';
-        fprintf (stderr,"%s\n", buff);
+        fprintf(stderr,"     :              .st_mtime) = %s",
+                               ctime(&status_buffer.st_mtime));
 
         /* creation time */
-        sprintf(buff,"     :              .st_ctime) = %s",
-                                        ctime(&status_buffer.st_ctime));
-        buff[57] = '\0';
-        fprintf (stderr,"%s\n", buff);
-
+        fprintf(stderr,"     :              .st_ctime) = %s",
+                               ctime(&status_buffer.st_ctime));
     } else {
         perror("FAIL ");
         return EXIT_FAILURE;
@@ -277,4 +261,3 @@
     return EXIT_SUCCESS;
 
 }
-
